code/datums/components/nanites.dm

```diff
@@ -104,10 +109,13 @@
 		var/datum/nanite_program/NP = X
 		NP.on_process()
 	set_nanite_bar()
-	if(cloud_id && world.time > next_sync)
+	if(cloud_id && cloud_active && world.time > next_sync)
 		cloud_sync()
 		next_sync = world.time + NANITE_SYNC_DELAY
 
+/datum/component/nanites/proc/delete_nanites()
+	qdel(src)

```

Comment:
wrappers for single procs are never worth it imo, just put needed destruction functionality in `Destroy` if you overrode this.

Reply:
i made this one to respond to the delete signal, but if i can just call qdel directly i might as well

Reply:
I'm actually not sure on that part, I'll have to ask.

Reply:
you don't seem to be able to pass params on signal registered procs so this is fine.